zerver/tests/test_signup.py

```diff
@@ -268,7 +268,7 @@ def test_register(self):
         with queries_captured() as queries:
             self.register(self.nonreg_email('test'), "test")
         # Ensure the number of queries we make is not O(streams)
-        self.assert_length(queries, 46)
+        self.assert_length(queries, 66)
```

Comment:
@rishig It's because of the calling of the `internal_send_private_message` in registration.py. If we remove that call and leave other commits intact its's 46. The moment the call is introduced a 20 extra queries result. But if we make another call to `internal_send_private_message` right after the first call it will increase the queries by 18. It's an increase of 18 for subsequent `internal_send_private_message` calls. 

I meant something like this when I mean subsequent `internal_send_provate_message` calls.

```py
def send_initial_pms(user):
    # type: (UserProfile) -> None
    content = '''Welcome to Zulip!
    This is a great place to test formatting, sending, and editing messages :pencil:.
    Click anywhere on this message to reply! A compose box will open at the bottom of the screen.'''

    internal_send_private_message(user.realm, get_system_bot(settings.WELCOME_BOT),
                                  user.email, content)
    internal_send_private_message(user.realm, get_system_bot(settings.WELCOME_BOT),
                                  user.email, content)
    internal_send_private_message(user.realm, get_system_bot(settings.WELCOME_BOT),
                                  user.email, content)

```
|No of internal_send_private_message calls| Query Count|
|---------------------------------------------------------|------------------|
|0 |46|
|1|66|
|2|82|
|3|98|
|4|114|

The  call to `internal_send_private_message` results in the invocation of different function calls like `recipient_for_emails` which does queries. I think we can account them as the reason for the increase in the queries.